app/controllers/concerns/cached_columns.rb

```diff
@@ -0,0 +1,21 @@
+module CachedColumns
+  CACHE_KEY = "%{identifier}/%{record_type}/column-keys/%{tail}".freeze
+
+  # Takes an ActiveRecord::Relation so we can make one more query off it
+  def cached_collection_column_keys(collection, identifier = "projects-#{@current_project.id}")
```

Comment:
This method name is working too hard, and still not being clear enough about what to expect from it. It doesn't return a key, it does work on a collection, but are columns really something we deal with?

Reply:
`cached_collection_column_keys` -> `collection_field_names`